### PR TITLE
feat: toml config and the new cli args for start

### PR DESCRIPTION
**Pull request by pimeys**
# Description

Closes: GB-6086

This introduces toml config parsing, and _just_ the parsing. We'll be plugging it into the engine in the next issue.

# Type of change

- [ ] 💔 Breaking
- [x] 🚀 Feature
- [ ] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [x] 🧪 Test
- [ ] 📦 Dependency
- [x] 📖 Requires documentation update

### CODE REVIEW

File: cli/crates/cli/src/start/configuration.rs [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+            allow_headers = ["😂😂😂"]
+        "#};
+
+        let error = toml::from_str::<Config>(input).unwrap_err();
+
+        insta::assert_snapshot!(&error.to_string(), @r###"
+        TOML parse error at line 2, column 17
+          |
+        2 | allow_headers = ["😂😂😂"]
+          |                 ^^^^^^^^^^^^^^^^
```
Comment (pimeys) [inline cli/crates/cli/src/start/configuration.rs:361]: Catched a serde bug here, see the ^^^'s :)

File: cli/crates/cli/src/start/configuration.rs [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+        "#};
+
+        let error = toml::from_str::<Config>(input).unwrap_err();
+
+        insta::assert_snapshot!(&error.to_string(), @r###"
+        TOML parse error at line 1, column 1
+          |
+        1 | [operation_limits]
+          | ^^^^^^^^^^^^^^^^^^
+        missing field `aliases`
```
Comment (tomhoule) [inline cli/crates/cli/src/start/configuration.rs:489]: Nitpick, but could we limit depth and height without limiting aliases? That seems to make sense

**Review by tomhoule — APPROVED** [review]
Smooth

**Comment by tomhoule** [comment]
Also it looks like the test breakage may be for real

File: cli/crates/cli/src/start/configuration/cors.rs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    pub expose_headers: Option<AnyOrAsciiStringArray>,
+    /// If set, allows browsers from private network to connect
+    #[serde(default)]
+    pub allow_private_network: bool,
+}
+
+#[derive(Debug, PartialEq, Clone, Copy, serde::Deserialize)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum HttpMethod {
+    Get,
```
Comment (jakubadamw) [inline cli/crates/cli/src/start/configuration/cors.rs:34]:
```rust
#[derive(derive_more::FromStr, derive_more::Display, serde_with::SerializeDisplay, serde_with::DeserializeFromStr)]
pub enum HttpMethod(http::Method);
```

**Review by jakubadamw — COMMENTED** [review]
Great! One thing, although not super important.

**Pull request merged**